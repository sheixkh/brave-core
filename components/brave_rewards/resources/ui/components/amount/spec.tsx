/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Amount from './index'
import { TestThemeProvider } from 'brave-ui/theme'

describe('Amount tests', () => {
  const doNothing = () => { console.log('nothing') }
  const baseComponent = (props?: object) => (
    <TestThemeProvider>
      <Amount id='amount' amount={'1'} converted={'0.4'} onSelect={doNothing} {...props} />
    </TestThemeProvider>
  )

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#amount').length
      expect(assertion).toBe(1)
    })
  })
})
