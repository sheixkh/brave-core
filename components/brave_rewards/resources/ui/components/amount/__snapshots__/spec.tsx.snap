// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount tests basic tests matches the snapshot 1`] = `
<button
  className="sc-bdVaJa eqkOmY"
  data-test-id="amount-wrapper"
  id="amount"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH kusHlF"
    type="small"
  >
    <div
      className="sc-ifAKCX kDLBtx"
    >
      <svg
        aria-hidden="true"
        className="sc-bAeIUo iUNCsN"
        focusable="false"
        viewBox="0 0 32 32"
      >
        <path
          d="M9.61 23.25h12.78L16 12 9.61 23.25z"
          fill="#fff"
          fillRule="evenodd"
        />
        <path
          d="M3 26.8l7.67-4.52L16 13V4a.45.45 0 0 0-.38.28l-6.27 11-6.26 11a.48.48 0 0 0 0 .48"
          fill="#ff4724"
          fillRule="evenodd"
        />
        <path
          d="M16 4v9l5.29 9.31L29 26.8a.48.48 0 0 0-.05-.48l-6.26-11-6.27-11A.45.45 0 0 0 16 4"
          fill="#9e1f63"
          fillRule="evenodd"
        />
        <path
          d="M29 26.8l-7.67-4.52H10.71L3 26.8a.47.47 0 0 0 .43.2h25.1a.47.47 0 0 0 .43-.2"
          fill="#662d91"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <span
      className="sc-bxivhb cDFCAd"
    >
      1
    </span>
     
    <div
      className="sc-htpNat eWCEFP"
    />
  </div>
  <div
    className="sc-EHOje kKXsJW"
    type="small"
  >
    MISSING: about
     
    0.4
     
    USD
  </div>
</button>
`;
